rust/kernel/fs.rs

```diff
@@ -44,6 +76,55 @@ impl<T: Type + ?Sized> Tables<T> {
         }
     }
 
+    unsafe extern "C" fn parse_param_callback(
```

Comment:
Doesn't all 'unsafe' comments require a '//SAFETY' comment ( #351 ) ?
 Although the following unsafe code it is indeed commented.

Reply:
Note that this is an `unsafe` block, but an `unsafe fn`. For the former, you indeed need to describe why the operations inside the block are sound with a `// SAFETY: ` comment. For the latter, we use `/// # Safety` documentation sections to describe the safety preconditions, if public.

It may be a good idea to later on start requiring documentation for private `unsafe fn` functions too, though.

Reply:
As Miguel explained, we normally require `/// # Safety` blocks for these, which is slightly different: we're telling callers what they need to do to use it safely. I do write these sections even for private functions (see for example `unnregister_keys` in this very same file), but this case is a callback from C: there are no Rust callers, so no one is going to write a `// SAFETY` block based on this function.

None of the C callbacks currently have `/// # Safety` sections because of this.